ouroboros-network/src/Ouroboros/Network/NodeToClient.hs

```diff
@@ -288,13 +290,23 @@ nodeToClientCodecCBORTerm = CodecCBORTerm {encodeTerm, decodeTerm}
     where
       encodeTerm :: NodeToClientVersionData -> CBOR.Term
       encodeTerm NodeToClientVersionData { networkMagic } =
-        CBOR.TInt (fromIntegral $ unNetworkMagic networkMagic)
-
-      decodeTerm :: CBOR.Term -> Either Text NodeToClientVersionData
-      decodeTerm (CBOR.TInt x) | x >= 0 && x <= 0xffffffff = Right (NodeToClientVersionData $ NetworkMagic $ fromIntegral x)
-                               | otherwise                 = Left $ T.pack $ "networkMagic out of bound: " <> show x
+        if (unNetworkMagic networkMagic) > maxInt32
+           then CBOR.TInt     (fromIntegral $ unNetworkMagic networkMagic)
+           else CBOR.TInteger (fromIntegral $ unNetworkMagic networkMagic)
+
+      decodeTerm (CBOR.TInt x) | x >= 0 && x <= maxInt32 = Right (NodeToClientVersionData $ NetworkMagic $ fromIntegral x)
+                               | otherwise    = Left $ T.pack $ "networkMagic out of bound: " <> show x
+      decodeTerm (CBOR.TInteger x) | x >= 0 && x <= maxWord32 = Right (NodeToClientVersionData $ NetworkMagic $ fromIntegral x)
+                               | otherwise                    = Left $ T.pack $ "networkMagic out of bound: " <> show x
       decodeTerm t             = Left $ T.pack $ "unknown encoding: " ++ show t
 
+      maxInt32 :: Integral a => a
```

Comment:
Do we need `maxInt32` and `maxWord32` to be polymorphic?

Reply:
I don't remember.